app/javascript/hooks/useIntl.ts

```diff
@@ -16,7 +16,7 @@ export type IntlShape = ParentIntlShape & {
 	formatMoney(amount: Money, opts?: FormatMoneyOptions): string;
 };
 
-export const IntlContext = React.createContext<IntlShape>(null as IntlShape);
+export const IntlContext = React.createContext<IntlShape>(null as unknown as IntlShape);
```

Comment:
Casting to unknown defeats the typechecker again. Sometimes its necessary but you should have a good justification.